Description: Add "Ubuntu" to the platform part of the UA string
Author: Chris Coulson <chris.coulson@canonical.com>
Forwarded: not-needed

--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -213,6 +213,9 @@
     , mLegacyAppName("Mozilla")
     , mLegacyAppVersion("5.0")
     , mProduct("Gecko")
+#ifdef MOZ_UA_VENDOR
+    , mVendor(MOZ_UA_VENDOR)
+#endif
     , mCompatFirefoxEnabled(false)
     , mUserAgentIsDirty(true)
     , mAcceptLanguagesIsDirty(true)
@@ -502,6 +505,9 @@
     LOG(("> misc = %s\n", mMisc.get()));
     LOG(("> product = %s\n", mProduct.get()));
     LOG(("> product-sub = %s\n", mProductSub.get()));
+#ifdef MOZ_UA_VENDOR
+    LOG(("> vendor = %s\n", mVendor.get()));
+#endif
     LOG(("> app-name = %s\n", mAppName.get()));
     LOG(("> app-version = %s\n", mAppVersion.get()));
     LOG(("> compat-firefox = %s\n", mCompatFirefox.get()));
@@ -872,8 +878,10 @@
                            mCompatFirefox.Length() +
                            mCompatDevice.Length() +
                            mDeviceModelId.Length() +
-                           13);
-
+#ifdef MOZ_UA_VENDOR
+                           mVendor.Length() +
+#endif
+                           15);
     // Application portion
     mUserAgent.Assign(mLegacyAppName);
     mUserAgent += '/';
@@ -888,6 +896,10 @@
       mUserAgent.AppendLiteral("; ");
     }
 #endif
+#ifdef MOZ_UA_VENDOR
+    mUserAgent += mVendor;
+    mUserAgent.AppendLiteral("; ");
+#endif
     if (!mCompatDevice.IsEmpty()) {
         mUserAgent += mCompatDevice;
         mUserAgent.AppendLiteral("; ");
--- a/netwerk/protocol/http/nsHttpHandler.h
+++ b/netwerk/protocol/http/nsHttpHandler.h
@@ -506,6 +506,7 @@
     // useragent components
     nsCString      mLegacyAppName;
     nsCString      mLegacyAppVersion;
+    nsCString      mVendor;
     nsCString      mPlatform;
     nsCString      mOscpu;
     nsCString      mMisc;
--- a/old-configure.in
+++ b/old-configure.in
@@ -2185,6 +2185,16 @@
     MOZ_APP_BASENAME="$WITH_APP_BASENAME"
 fi
 
+# Allow someone to add a vendor component to the default user agent string
+MOZ_ARG_WITH_STRING(ua-vendor,
+[--with-ua-vendor=VENDOR sets MOZ_UA_VENDOR to VENDOR],
+WITH_UA_VENDOR=$withval,
+)
+
+if test -n "$WITH_UA_VENDOR" ; then
+    MOZ_UA_VENDOR="$WITH_UA_VENDOR"
+fi
+
 # Special cases where we need to AC_DEFINE something. Also a holdover for apps
 # that haven't made a confvars.sh yet. Don't add new stuff here, use
 # MOZ_BUILD_APP.
@@ -4744,6 +4754,7 @@
 AC_SUBST(MOZ_APP_DISPLAYNAME)
 AC_SUBST(MOZ_APP_BASENAME)
 AC_SUBST(MOZ_APP_VENDOR)
+AC_SUBST(MOZ_UA_VENDOR)
 AC_SUBST(MOZ_APP_PROFILE)
 AC_SUBST(MOZ_APP_ID)
 AC_SUBST(MOZ_APP_ANDROID_VERSION_CODE)
--- a/build/moz.configure/old.configure
+++ b/build/moz.configure/old.configure
@@ -278,6 +278,7 @@
     '--with-system-zlib',
     '--with-thumb',
     '--with-thumb-interwork',
+    '--with-ua-vendor',
     '--with-unify-dist',
     '--with-user-appdir',
     '--x-includes',
--- a/netwerk/protocol/http/moz.build
+++ b/netwerk/protocol/http/moz.build
@@ -138,3 +138,6 @@
         DEFINES.update(
             HAS_CONNECTX=True,
         )
+
+if CONFIG['MOZ_UA_VENDOR']:
+    DEFINES['MOZ_UA_VENDOR'] = '"%s"' % CONFIG['MOZ_UA_VENDOR']
